### PR TITLE
Validate SAN trust domain during mTLS

### LINKED ISSUE #359: Verify inbound certificates properly
Currently, we only check the full SAN with a authz policy. otherwise we only check the root cert.

We also need to check the trust domain matches to get parity with Envoy's enforcement.

We should also double check (ie actually test) the root cert check actually happens, just in case - but it does appear to be there.

### PR DESCRIPTION
**Pull request by keithmattix**
Fixes #359

### CODE REVIEW

File: src/tls/boring.rs [hunk 1]
```diff
@@ -411,9 +419,31 @@ impl Verifier {
         cert.verify_san(identity)
     }
 
+    fn verifiy_san_trust_domain(&self, ctx: &mut X509StoreContextRef) -> Result<(), TlsError> {
+        let Self::SanTrustDomain(identity) = self else {
```
Comment (howardjohn) [inline src/tls/boring.rs:423]: nit: I see this is copied from above but its kind of a weird pattern. If we are always getting SanTrustDomain's in this, we should just pass the inner identity probably and not take `self`?

**Review by howardjohn — APPROVED** [review]
A test for this would be nice. I think we can do it with a test like test_hbone_ip_mismatch

**Comment by keithmattix** [comment]
Yeah, I plan to write a test, but I was running into a "not implemented" error locally and wanted to see if it was just my machine. Prow is reporting it too, so I think there's a mock I need to update somewhere

File: src/tls/boring.rs (hunk 1 above)
Comment (hzxuzhonghu) [reply src/tls/boring.rs:423]: And this match is redundant, guranteed in the caller

Comment (keithmattix) [reply src/tls/boring.rs:426]: I didn't get a chance to refactor this before the PR auto-merged 😢

**Pull request merged**